pkg/cli/admin/inspect/discovery.go

```diff
@@ -0,0 +1,56 @@
+package inspect
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/json"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/yaml"
+
+	"k8s.io/client-go/discovery"
+)
+
+func InspectDiscovery(ctx context.Context, destDir string, discoveryClient discovery.CachedDiscoveryInterface) error {
+	if err := writeDiscovery(ctx, filepath.Join(destDir, "discovery-apis.yaml"), discoveryClient, "/apis"); err != nil {
+		return err
+	}
+	if err := writeDiscovery(ctx, filepath.Join(destDir, "discovery-api.yaml"), discoveryClient, "/api"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func writeDiscovery(ctx context.Context, destFile string, discoveryClient discovery.CachedDiscoveryInterface, url string) error {
+	var responseContentType string
+	var statusCode int
+	apiBytes, err := discoveryClient.RESTClient().Get().
+		AbsPath(url).
+		SetHeader("Accept", discovery.AcceptV2).
+		Do(ctx).
+		ContentType(&responseContentType).
+		StatusCode(&statusCode).
+		Raw()
+	if err != nil {
+		return fmt.Errorf("discovery %q failed: %w", url, err)
+	}
+
+	if statusCode < 200 && statusCode > 299 {
```

Comment:
Why we are not checking only `statusCode != http.StatusOK`. Since we are not sending any etags (may result in 304), the only expected status should be 200?

Reply:
> Why we are not checking only `statusCode != http.StatusOK`. Since we are not sending any etags (may result in 304), the only expected status should be 200?

I'm not completely sure how the server responds and figured anything in the "ok" range was probably ok